### DISCUSSION TITLE
Modal / Dialog and React 18 Strict Mode Compatibility

### DISCUSSION THREAD

**Discussion by SimTimms** [adobe/react-spectrum#3602] [Q&A] [answered]
Hi,

I'm attempting to use the Dialog component for a technical test that Adobe have asked me to complete. While attempting to use the Dialog component with React 18 I receive this message:

```
Warning: Legacy context API has been detected within a strict-mode tree.

The old API will be supported in all 16.x releases, but applications using it should migrate to the new version.

Please update the following components: Transition
```

This issue has been raised before [https://github.com/adobe/react-spectrum/issues/1135](https://github.com/adobe/react-spectrum/issues/1135) but I can't find a solution.

I could remove StrictMode or revert to React 17 but as it's a technical test this might seem like a hack rather than a solution.

Does anyone know of a fix for this issue?

Many Thanks

Tim

**Comment by reidbarber** [comment, marked as answer]
React Spectrum doesn't currently support React Strict Mode yet, so I would suggest removing Strict Mode from your application. I don't think there is a workaround, and I think that is better than downgrading React.

**Reply by SimTimms** [reply]
Thank you for the quick response!
